crates/llms/src/chunking/mod.rs

```diff
@@ -38,37 +38,38 @@ pub trait Chunker: Sync + Send {
     }
 }
 
```

Comment:
The shared trait behind these `text_splitter::Splitter` is parameterised as follows
```rust
pub trait Splitter<Sizer>
where
    Sizer: ChunkSizer,
{
    type Level: SemanticLevel;
```

Implementations of `Splitter` do not share a `SemanticLevel` in common, making it hard to use
```rust
pub struct CharacterSplittingChunker {
    splitter: Arc<dyn text_splitter::Splitter>
}
```